src/utils/arithmetic.js

```diff
@@ -0,0 +1,17 @@
+import { ERROR_TEXT, INPUT_NOT_ALLOWED_MESSAGE } from '../constants/error';
+
+export const arithmetic = (augend, addend = 0, operation) => {
+  switch (operation) {
+    case '+':
+      return augend + addend;
+    case '-':
+      return augend - addend;
+    case '/':
+      return Math.floor(augend / addend);
+    case 'X':
+      return augend * addend;
+    default:
+      alert(INPUT_NOT_ALLOWED_MESSAGE);
+      return ERROR_TEXT;
+  }
```

Comment:
```suggestion
const calculator = {
  '+': (a: number, b: number) => a + b,
  '-': (a: number, b: number) => a - b,
  X: (a: number, b: number) => a * b,
  '/': (a: number, b: number) => Math.floor(a / b),
}

const result = calculator[operation]?.(augend, addend) 
if (result == null) {
 alert ...
```

이런식으로 switch"문"이 아니라 함수 호출과 같은 표현"식" 으로 작성하는 방법도 있습니다.
제어문은 논리의 흐름을 따라가기 복잡해지는 반면, 식으로 작성하면 항상 새로운 값을 생성해내기 때문에, 
흐름을 따라가기가 간편해집니다 😄 

처음에 질문주셨던 `else와 중첩 if 사용에 대하여...` 에 대한 해법 중의 하나도 가능하면 표현식을 사용한다 가 있습니다.
(극악의 삼항연산자로 다 도배해버리면 그건 그것대로 또 문제이지만요.. 😵 )


(참고: https://github.com/next-step/react-calculator/pull/25#discussion_r1103820695)

Reply:
항상 여러 케이스가 존재하는 로직에 대해서 switch를 많이 사용했는데
위 같은 방식으로 진행해보니 사용하는 측에서도 코드를 직관적 볼 수 있어서 좋았습니다!